### PR TITLE
Also search for STIX2 files in directories in MVT_STIX2

### PR DESCRIPTION
**Pull request by besendorf**
The old version of this method only accepted paths to files in MVT_STIX2. This new version checks whether a variable in MVT_STIX2 is a file or a dir and then recursively searches for .stix2 files in those dirs.

A change I made is that it only processes .stix2 files which I thought is a reasonably check.

### CODE REVIEW

File: mvt/common/indicators.py [hunk 1]
```diff
@@ -40,15 +40,20 @@ def _load_downloaded_indicators(self) -> None:
...
 
         paths = os.environ["MVT_STIX2"].split(":")
         for path in paths:
-            if os.path.isfile(path):
+            if os.path.isfile(path) and path.lower().endswith(".stix2"):
                 self.parse_stix2(path)
+            if os.path.isdir(path):
+                for root, dirs, files in os.walk(path):
+                    for filename in files:
+                        if filename.lower().endswith(".stix2"):
```
Comment (roaree) [inline mvt/common/indicators.py:55]: Non-blocker: we might have some examples of using glob that could condense this into one for loop

**Review by roaree — APPROVED** [review]
Looks great! thanks

File: mvt/common/indicators.py (hunk 1 above)
Comment (besendorf) [reply mvt/common/indicators.py:55]: happy to change that if you can show me an example.

Comment (Te-k) [reply mvt/common/indicators.py:55]: I think he means something like [this](https://github.com/mvt-project/mvt/blob/efe46d7b49d383d97151e03bbde3da6970f00f5f/mvt/ios/decrypt.py#L134) using the [glob library](https://docs.python.org/3/library/glob.html)

Comment (DonnchaC) [reply mvt/common/indicators.py:55]: > I think he means something like [this](https://github.com/mvt-project/mvt/blob/efe46d7b49d383d97151e03bbde3da6970f00f5f/mvt/ios/decrypt.py#L134) using the [glob library](https://docs.python.org/3/library/glob.html)

Yes exactly. The glob makes it easier to rationalise about what the code is doing IMO.

Comment (besendorf) [reply mvt/common/indicators.py]: I changed the code to use glob.glob instead. 
note: this is case-sensitive now. Case insensitivity with glob was only added in python3.12

**Pull request merged**